src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
```

Comment:
```suggestion
  const grouped: GroupsMap<T> = {};
```
I'm sure that mentor will ask to rename it in clearer way :) Optional change.